lib/hexo/router.js

```diff
@@ -3,18 +3,63 @@
 const { EventEmitter } = require('events');
 const Promise = require('bluebird');
 const Stream = require('stream');
-const util = require('util');
+const { inherits } = require('util');
 const { Readable } = Stream;
 
-function Router() {
-  Reflect.apply(EventEmitter, this, []);
+class RouteStream {
+  constructor(data) {
+    Reflect.apply(Readable, this, [{objectMode: true}]);
 
-  this.routes = {};
+    this._data = data.data;
+    this._ended = false;
+    this.modified = data.modified;
+  }
+
+  _read() {
+    const data = this._data;
+
+    if (typeof data !== 'function') {
+      this.push(data);
+      this.push(null);
+      return;
+    }
+
+    // Don't read it twice!
+    if (this._ended) return false;
+    this._ended = true;
+
+    data().then(data => {
+      if (data instanceof Stream && data.readable) {
+        data.on('data', d => {
+          this.push(d);
+        });
+
+        data.on('end', () => {
+          this.push(null);
+        });
+
+        data.on('error', err => {
+          this.emit('error', err);
+        });
+      } else if (data instanceof Buffer || typeof data === 'string') {
+        this.push(data);
+        this.push(null);
+      } else if (typeof data === 'object') {
+        this.push(JSON.stringify(data));
+        this.push(null);
+      } else {
+        this.push(null);
+      }
+    }).catch(err => {
+      this.emit('error', err);
+      this.push(null);
+    });
+  }
 }
 
-util.inherits(Router, EventEmitter);
+inherits(RouteStream, Readable);
```

Comment:
Why you don't use `extends`?

```js
class RouteStream extends Readable {
  constructor(data) {
    super();
```

Reply:
still new to Class syntax, thanks for the heads up. I'll apply `extends` to other PRs.